src/webui/www/private/scripts/client.js

```diff
@@ -768,22 +793,54 @@ window.addEvent('load', function() {
     $('showSearchEngineLink').addEvent('click', function(e) {
         showSearchEngine = !showSearchEngine;
         LocalPreferences.set('show_search_engine', showSearchEngine.toString());
+        initializeTabs();
+    });
+
+    $('showRssReaderLink').addEvent('click', function(e) {
+        showRssReader = !showRssReader;
+        LocalPreferences.set('show_rss_reader', showRssReader.toString());
+        initializeTabs();
```

Comment:
I'm not sure, will this work? `initializeTabs` is defined later.

Reply:
Works because `initializeTabs` isn't actually called there. It's only called on click so after initialization.

Though since there was quite a bit of duplicate code I moved things around and renamed the function to `updateTabDisplay` to better reflect its purpose.

I also removed the adding and removing of the tabs event listeners , I think it's fine to have those three event listeners even when the buttons are not displayed. Please take a look and give feedback.